src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -401,7 +410,14 @@ private function getId($key)
     {
         CacheItem::validateKey($key);
 
-        return $this->namespace.$key;
+        if (null === $this->maxIdLength) {
+            return $this->namespace.$key;
+        }
+        if (strlen($id = $this->namespace.$key) > $this->maxIdLength) {
+            $id = $this->namespace.substr_replace(base64_encode(md5($key, true)), ':', -2);
```

Comment:
The added trailing `:` is to prevent any collision between hashed keys and non-hasked keys.